pkg/monitors/kubernetes/volumes/pvdim.go

```diff
@@ -64,17 +67,32 @@ func dimsForPersistentVolumeSource(pvs v1.PersistentVolumeSource) map[string]str
 // specified volumes and those through PersistentVolumes so we have to
 // accommodate both types even though they have largely the same members.
 func dimsForVolumeSource(vs v1.VolumeSource, namespace string, client *k8s.Clientset) (map[string]string, error) {
-	// IF YOU ADD A NEW TYPE HERE, ADD IT IN dimsForPersistentVolumeSource
+	// IF YOU ADD A NEW PERSISTENT TYPE HERE, ADD IT IN dimsForPersistentVolumeSource
 	// ABOVE TOO!
 	// PersistentVolumeClaim is unique to VolumeSource, PersistentVolumeSource
 	// will not have this.
 	switch {
+	// Special case for PVC as it will invoke dimsForPersistentVolumeSource lower in the chain:
 	case vs.PersistentVolumeClaim != nil:
 		return dimsForPersistentVolumeClaim(vs.PersistentVolumeClaim.ClaimName, namespace, client)
+
+	// PERSISTENT TYPES:
 	case vs.AWSElasticBlockStore != nil:
 		return awsElasticBlockStoreDims(*vs.AWSElasticBlockStore), nil
+	case vs.GCEPersistentDisk != nil:
+		return gcePersistentDiskDims(*vs.GCEPersistentDisk), nil
 	case vs.Glusterfs != nil:
 		return glusterfsDims(*vs.Glusterfs), nil
+
+	// NON PERSISTENT TYPES:
+	case vs.ConfigMap != nil:
+		return map[string]string{"volume_type": "configMap"}, nil
```

Comment:
Not 100% about which case to adopt here.

Reply:
I would do `ConfigMap` since that is so prevalent in K8s.

Reply:
Re-reading the docs about the `volume_type` dimension (https://github.com/signalfx/signalfx-agent/blob/v5.3.1/pkg/monitors/kubernetes/volumes/metadata.yaml#L22-L24):

> The type of the underlying volume -- this will be *the key used in the k8s volume config spec* (e.g. awsElasticBlockStore, etc.)

and the mentioned the volume spec reference (https://kubernetes.io/docs/concepts/storage/volumes/#configmap), it turns out `configMap` is probably better after all.

Reply:
Ok, sounds good!